### PR TITLE
[sdk] Update to Gecko SDK 4.2.0

### PR DESCRIPTION
**Pull request by lmnotran**
This updates the `gecko_sdk` submodule to `4.2.0` and also updates the PAL in `src/src`

### Known Issues
- `src/src/crypto.c` has changes in GSDK 4.2.0 that are not pulled into this set of changes. The changes depend on some openthread stack changes which have not been upstreamed from the GSDK to openthread/openthread yet. Trying to build with the `crypto.c` from GSDK 4.2.0 leads to this error because of the changes which haven't been upstreamed yet 
```
/Users/matran/repos/ot-efr32/src/src/crypto.c: In function 'otPlatCryptoEcdsaGenerateKey':
/Users/matran/repos/ot-efr32/src/src/crypto.c:420:50: error: 'otPlatCryptoEcdsaKeyPair' has no member named 'mKeyRef'
  420 |     status = sl_sec_man_generate_key(   &aKeyPair->mKeyRef,
      |                                                  ^~
/Users/matran/repos/ot-efr32/src/src/crypto.c: In function 'otPlatCryptoEcdsaGetPublicKey':
/Users/matran/repos/ot-efr32/src/src/crypto.c:444:53: error: 'otPlatCryptoEcdsaKeyPair' has no member named 'mKeyRef'
  444 |     status = sl_sec_man_export_public_key(  aKeyPair->mKeyRef,
      |                                                     ^~
/Users/matran/repos/ot-efr32/src/src/crypto.c: In function 'otPlatCryptoEcdsaSign':
/Users/matran/repos/ot-efr32/src/src/crypto.c:467:41: error: 'otPlatCryptoEcdsaKeyPair' has no member named 'mKeyRef'
  467 |     status = sl_sec_man_sign(   aKeyPair->mKeyRef,
      |                                         ^~
```

### CODE REVIEW

**Comment by parag-silabs** [comment]
@lmnotran Thanks for removing a patch file`third_party/silabs/patches/0001-Remove-stack-components-to-prevent-adding-OpenThread.patch`. Should we clean up`/script/build`? Where we call apply_patches and revert_patches for rcp_spi target.

**Comment by lmnotran** [comment]
> @lmnotran No changes to crypto.c?

@hemanth-silabs Nope! Pulling in crypto.c changes causes build failures (See PR description)

**Review by suveshpratapa — CHANGES_REQUESTED** [review]
@lmnotran We need to pull in platform-abstraction/posix/openthread-core-silabs-posix-config.h somewhere in here.  This is now a file we provide for those wishing to apply SiLabs configs to their OTBR.  It's part of the GSDK offering, so I assume we also need to put it in here somewhere.  I'll leave the decision of the right location to you.

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by hemanth-silabs — APPROVED** [review]
LGTM 👍

File: src/src/sl_rcp_gp_interface.c [hunk 1]
```diff
@@ -0,0 +1,235 @@
+/*******************************************************************************
```
Comment (romacdon) [inline src/src/sl_rcp_gp_interface.c:1]: Is this the correct license header for stuff committed to the openthread/ot-efr32 GitHub repo?

**Review by romacdon — CHANGES_REQUESTED** [review]

File: src/src/sl_rcp_gp_interface.c (hunk 1 above)
Comment (lmnotran) [reply src/src/sl_rcp_gp_interface.c:1]: Good catch. Lemme fix that quick

**Review by romacdon — APPROVED** [review]
LGTM

**Review by bertoldi-silabs — APPROVED** [review]

**Comment by lmnotran** [comment]
@jwhui can you review this please?

**Comment by romacdon** [comment]
@jwhui any chance we can get this approved and merged today?  Thanks

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@lmnotran , can you help resolve conflicts? Thanks!

**Comment by lmnotran** [comment]
@jwhui Should be good to go now. Thanks for reviewing!

**Pull request merged**